src/HttpRequest.h

```diff
@@ -146,13 +149,38 @@ class HttpRequest: public Http::Message
 
     int imslen;
 
-    Ip::Address client_addr;
+    /// mark this request as initiated by Squid (rather than received on a client connection)
+    void selfInitiated();
+
+    /// whether this request was initiated by Squid (rather than received on a client connection)
+    bool isSelfInitiated() const { return selfInitiated_; }
```

Comment:
After selfInitiated() above is fixed:

```suggestion
    bool selfInitiated() const { return selfInitiated_; }
```

Reply:
Done.